src/main/scala/io/iohk/ethereum/db/dataSource/DataSource.scala

```diff
@@ -24,6 +24,17 @@ trait DataSource {
     */
   def get(namespace: Namespace, key: Key): Option[Value]
 
+
+  /**
+    * This function obtains the associated value to a key, if there exists one. It assumes that
+    * caller already properly serialized key. Useful when caller knows some pattern in data to
+    * avoid generic serialization.
+    *
+    * @param key
+    * @return the value associated with the passed key.
+    */
+  def getOptimized(key: Array[Byte]): Option[Array[Byte]]
```

Comment:
I'm not quite sure I follow where the improvement was made here :thinking:

Reply:
Normal data source `get` has as an argument `IndexedSeq[Byte]`, and prepends `namespace` byte to it and converts it `Array[Byte]`, also it returns `IndexedSeq[Byte]` which need further deserialization. This one assumes that all of this was already done and takes `Array[Byte]`. Also it returns `Array[Byte]` to

Out node storage is most busy of all storages and this constant conversions just started showing on profiling.
This is especially true for value, which serializtion route is `Array[Byte] -> IndexedSeq[Byte] -> Array[Byte]`

Reply:
What do you think about moving that logic to the `DataSource` as a `get` with a different signature?

Reply:
it is defined in `DataSource`, I named it differently hoping that it will be little explanation why we have two get with different signatures. You think just renaming it would be more clear?

Reply:
I think of _namespaces_ as some kind of column families and it's related to the DataSource. That's why i'm not quite sure it's ok to bypass that at storage level.

Reply:
I agree that it is not pretty and breaks our boundaries between `storages` and `dataSource` that why I named methods in data source differently and added docs to them.

 `    * This function obtains the associated value to a key, if there exists one. It assumes that
    * caller already properly serialized key. Useful when caller knows some pattern in data to
    * avoid generic serialization.`

Maybe some caution in those docs about this namespace and boundry breaking, would be helpful? i mean in other storages this change wont make much sense as they not used as much as `nodeStorage` especially when

Reply:
What about this? To avoid turning this into an endless discussion, lets move forward as it it and let's add a task in youtrack to do a second analysis round about this and maybe do a refactor if necessary.

Reply:
I have created task in yt to not get it lost